src/commands/appSettings/SqlDatabaseConnectionPromptStep.ts

```diff
@@ -0,0 +1,92 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzureWizardPromptStep, ISubscriptionActionContext, IWizardOptions } from '@microsoft/vscode-azext-utils';
+import { MessageItem } from 'vscode';
+import { ConnectionKey, ConnectionType } from '../../constants';
+import { skipForNow } from '../../constants-nls';
+import { ext } from '../../extensionVariables';
+import { getLocalConnectionString } from '../../funcConfig/local.settings';
+import { localize } from '../../localize';
+import { SqlServerListStep } from '../createFunction/durableSteps/sql/SqlServerListStep';
+import { IConnectionPromptOptions } from './IConnectionPromptOptions';
+import { ISqlDatabaseConnectionWizardContext } from './ISqlDatabaseConnectionWizardContext';
+import { SqlDatabaseConnectionCustomPromptStep } from './SqlDatabaseConnectionCustomPromptStep';
+
+export class SqlDatabaseConnectionPromptStep<T extends ISqlDatabaseConnectionWizardContext> extends AzureWizardPromptStep<T> {
+    public constructor(private readonly _options?: IConnectionPromptOptions) {
+        super();
+    }
+
+    public async prompt(context: T): Promise<void> {
+        const connectAzureDatabase: MessageItem = { title: localize('connectSqlDatabase', 'Connect Azure SQL Database') };
+        const connectNonAzureDatabase: MessageItem = { title: localize('connectSqlDatabase', 'Connect Non-Azure SQL Database') };
+        const useExistingConnectionButton: MessageItem = { title: localize('useExistingConnection', 'Use Existing Connection') };
+        const skipForNowButton: MessageItem = { title: skipForNow };
+
+        const message: string = localize('selectSqlDatabaseConnection', 'In order to proceed, you must connect a SQL database for internal use by the Azure Functions runtime.');
+
+        const buttons: MessageItem[] = [connectAzureDatabase, connectNonAzureDatabase];
+
+        if (!this._options?.suppressSkipForNow) {
+            buttons.push(skipForNowButton);
+        } else if (this._options?.suppressSkipForNow && !context.sqlDbRemoteConnection) {
+            // On debug, give user the option to run from an existing local connection string
```

Comment:
I think we had talked about this before (when removing `skipForNow` was also talked about) and agreed it would be best to try and have the prompts converge to the `emulator` equivalent scenario, upon which they would no longer prompt during debug.  I was going to address this convergence logic during the debug PR since this also needs to be addressed across all scenarios including `azureWebJobsStorage` and `netherite`.  Are you opposed to delaying this change for the future debug PR?  I've added the issue here for tracking: https://github.com/microsoft/vscode-azurefunctions/issues/3471

Reply:
Oh nah, we'll skip this comment for now. 🥁